### PR TITLE
docs: add FHIR overview guide and extensions example

### PR DESCRIPTION
**Pull request by Goncharo**
refs. metriport/metriport#578

### Dependencies

N/A

### Description

add FHIR overview guide and extensions example

### Release Plan

asap

### CODE REVIEW

File: docs/medical-api/fhir/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+## Resources & Data Types
+
+After you initiate a new document query for a given patient, Metriport will automatically convert
+any newly found C-CDA documents from XML to the corresponding FHIR Resources in JSON.
+
+A FHIR Resource is a representation of some concept related to healthcare. Some examples include:
+
+- [Encounter](/medical-api/fhir/resources/encounter): an interaction between a patient and healthcare provider(s).
+- [MedicationRequst](/medical-api/fhir/resources/medicationrequest): an order or request for both supply of the medication and the instructions for administration of the medication to a patient.
```
Comment (leite08) [inline docs/medical-api/fhir/overview.mdx:26]: typo: `MedicationRequst` > `MedicationRequest`

File: docs/medical-api/fhir/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    "system": "http://unitsofmeasure.org",
+    "code": "cm"
+  }
+}
+```
+
+You're able to query for a Patient's FHIR Resources using the [Get Patient's Consolidated Data
+Endpoint](/medical-api/api-reference/fhir/get-patient-consolidated).
+
+Additionally, to meet data contribution requirements, you're able to provide new data about a
```
Comment (leite08) [inline docs/medical-api/fhir/overview.mdx:112]: Wondering if we could have a link on `data contribution requirements` to the description about that? It's good to have product people involved on our requirements, they might not be aware of requirements since they're not the ones signing contracts, in most cases.

File: docs/medical-api/api-reference/fhir/add-patient-data.mdx
Comment (leite08) [inline docs/medical-api/api-reference/fhir/add-patient-data.mdx:1]: We don't have any schema/DTO structure currently that we could add to the docs?

**Review by leite08 — APPROVED** [review]
LGTM, just a few comments/suggestions

File: docs/medical-api/api-reference/fhir/add-patient-data.mdx
Comment (Goncharo) [reply docs/medical-api/api-reference/fhir/add-patient-data.mdx:1]: Until we're ready to ship this publicly, I'm not sure there's a benefit adding anything that's a WIP here - could just add confusion if we decide to change it.

File: docs/medical-api/fhir/overview.mdx (hunk 1 above)
Comment (Goncharo) [reply docs/medical-api/fhir/overview.mdx:26]: fixed

File: docs/medical-api/fhir/extensions/doc-id.mdx
[the remark this answers is not in the document]
Comment (Goncharo) [reply docs/medical-api/fhir/extensions/doc-id.mdx:1]: There's currently no way to query for FHIR resources corresponding to an extension value - only the other way around.

For this we would have to customize the FHIR server for custom search params.

File: docs/medical-api/fhir/overview.mdx (hunk 2 above)
Comment (Goncharo) [reply docs/medical-api/fhir/overview.mdx:112]: Will reword this to avoid overhead & another link - FYI this requirement is in the ToS

Comment (Goncharo) [reply docs/medical-api/fhir/overview.mdx:112]: Makes sense - and we should probably describe this in more detail in general: ticket here https://github.com/metriport/metriport/issues/597
